File: C++/source/Framework/Messaging/LightHouse/LightHouseTransportObjectManager.cpp
/***************************************************************************
 *   Copyright (C) 20016 Vidyasagara Guntaka                               *
 *   All rights reserved.                                                  *
 *   Author : Vidyasagara Reddy Guntaka                                    *
 ***************************************************************************/

#include "Framework/Messaging/LightHouse/LightHouseTransportObjectManager.h"
#include "Framework/Messaging/LightHouse/LightHouseTransportBroadcastLightPulseWorker.h"

namespace WaveNs
{

LightHouseTransportObjectManager::LightHouseTransportObjectManager ()
    : WaveLocalObjectManager (getServiceName ())
{
    m_pLightHouseTransportBroadcastLightPulseWorker = new LightHouseTransportBroadcastLightPulseWorker (this);

    waveAssert (NULL != m_pLightHouseTransportBroadcastLightPulseWorker, __FILE__, __LINE__);
}

LightHouseTransportObjectManager::~LightHouseTransportObjectManager ()
{
    if (NULL != m_pLightHouseTransportBroadcastLightPulseWorker)
    {
       delete m_pLightHouseTransportBroadcastLightPulseWorker;
    }
}

string LightHouseTransportObjectManager::getServiceName ()
{
    return ("Light House Transport");
}

LightHouseTransportObjectManager *LightHouseTransportObjectManager::getInstance ()
{
    static LightHouseTransportObjectManager *pLightHouseTransportObjectManager = new LightHouseTransportObjectManager ();

    WaveNs::waveAssert (NULL != pLightHouseTransportObjectManager, __FILE__, __LINE__);

    return (pLightHouseTransportObjectManager);
}

WaveServiceId LightHouseTransportObjectManager::getWaveServiceId ()
{
    return ((getInstance ())->getServiceId ());
}

}

File: Java/source/com/CxWave/Wave/Framework/Messaging/LightHouse/LightHouseTransportObjectManager.java
/*************************************************************************************************************************
 * Copyright (C) 2015-2016 Vidyasagara Guntaka & CxWave, Inc * All rights reserved. * Author : Vidyasagara Reddy Guntaka *
 *************************************************************************************************************************/

package com.CxWave.Wave.Framework.Messaging.LightHouse;

import com.CxWave.Wave.Framework.ObjectModel.WaveLocalObjectManager;
import com.CxWave.Wave.Framework.Type.WaveServiceId;
import com.CxWave.Wave.Framework.Utils.Assert.WaveAssertUtils;

public class LightHouseTransportObjectManager extends WaveLocalObjectManager
{
    private static LightHouseTransportObjectManager s_lightHouseTransportObjectManager = null;

    public LightHouseTransportObjectManager ()
    {
        super (getServiceName ());
    }

    public static String getServiceName ()
    {
        return ("Light House Transport");
    }

    public static LightHouseTransportObjectManager getInstance ()
    {
        if (null == s_lightHouseTransportObjectManager)
        {
            s_lightHouseTransportObjectManager = new LightHouseTransportObjectManager ();
        }

        WaveAssertUtils.waveAssert (null != s_lightHouseTransportObjectManager);

        return (s_lightHouseTransportObjectManager);
    }

    public static WaveServiceId getWaveServiceId ()
    {
        return ((getInstance ()).getServiceId ());
    }
}
